<?xml version="1.0"?>
<robot  xmlns:xacro="http://www.ros.org/wiki/xacro">
        <xacro:property name="base_length" value="0.6"/>
        <xacro:property name="base_width" value="0.4"/>
        <xacro:property name="base_height" value="0.2"/>
        <xacro:property name="wheel_radius" value="0.1"/>
        <xacro:property name="wheel_length" value="0.05"/>
  
    
        
        
        <link name="base_foot"/>  
    <joint name="base" type="fixed">
       <parent link="base_foot"/>
       <child link="base_link"/>
       <origin xyz="0 0 ${wheel_radius}" rpy="0 0 0"/>
    </joint>    
        <link name="base_link">
            <visual>
                <geometry>
                    <box size="${base_length} ${base_width} ${base_height}"/>
                </geometry>
                <origin xyz="0 0 0" rpy="0 0 0"/>
                <material name="green"/>
            </visual>
            <collision>
                <geometry>
                    <box size="${base_length} ${base_width} ${base_height}"/>
                </geometry>
                <origin xyz="0 0 0" rpy="0 0 0"/>
            </collision>
            <xacro:box_inertia m="5.0" l="${base_length}" w="${base_width}" h="${base_height}" xyz="0 0 0" rpy="0 0 0"/>
        </link>
        
        <xacro:macro name="wheel_link" params ="prefix">
           <link name="${prefix}_link" >
           	<visual>
                <geometry>
                    <cylinder radius="${wheel_radius}" length="${wheel_length}"/>
                </geometry>
                <origin xyz="0 0 0" rpy="${pi/2.0} 0 0"/>
                <material name ="gray"/>
            </visual>
            <collision>
                <geometry>
                    <cylinder radius="${wheel_radius}" length="${wheel_length}"/>
                </geometry>
                <origin xyz="0 0 0" rpy="${pi/2.0} 0 0"/>
            </collision>
            <xacro:cylinder_inertia m="1.0" r="${wheel_radius}" h="${wheel_length}" 
                                    xyz="0 0 0" rpy="${pi / 2.0} 0 0" />
           </link>
        </xacro:macro>
        
        <xacro:wheel_link prefix="right" />
        <xacro:wheel_link prefix="left"  />
        
    <joint name="right_joint" type="continuous">
        <parent link="base_link"/>
        <child link="right_link"/>
        <origin xyz="${(-base_length/6.0)} ${-(base_width + wheel_length)/2.0} ${-base_height/2.0}" rpy="0 0 0"/>
        <axis xyz="0 1 0"/>
    </joint>
       
    <joint name="left_joint" type="continuous">
       <parent link="base_link"/>
       <child link ="left_link"/>
       <origin xyz="${(-base_length/6.0)} ${(base_width + wheel_length)/2.0} ${(-base_height/2.0)}" rpy="0 0 0"/>
       <axis xyz="0 1 0"/>
    </joint>
        <link name="caster_wheel">
            <visual>
                <geometry>
                    <sphere radius="${(wheel_radius/2.0)}"/>
                </geometry>
                <origin xyz="0 0 0" rpy="0 0 0"/>
                <material name="gray"/>
            </visual>
            <collision>
                <geometry>
                    <sphere radius="${(wheel_radius/2.0)}"/>
                </geometry>
                <origin xyz="0 0 0" rpy="0 0 0"/>
            </collision>
            <xacro:sphere_inertia m="0.5" r="${wheel_radius / 2.0}"
                              xyz="0 0 0" rpy="0 0 0" />
        </link>
    <joint name="caster joint" type="fixed"> 
        <parent link="base_link"/>
        <child link="caster_wheel"/>
        <origin xyz="${(base_length/3.0)} 0 ${-(wheel_radius + wheel_length)}" rpy="0 0 0"/>
        
    </joint>
    <!-- <joint name="lider_base" type="revolute">
        <parent link="base_link"/>
        <child link="lider_link"/>
        <origin xyz="0 0 0.125" rpy="0 0 0"/>
        <axis xyz="0 0 1"/>
        <limit lower="${(-pi)}" upper="${(pi)}" velocity="100" effort="100"/>

    </joint>
       <link name="lider_link">
            <visual>
                <geometry>
                    <cylinder radius="${(wheel_radius)}" length="${(wheel_length)}"/>
                </geometry>
                <origin xyz="0 0 0" rpy="0 0 0"/>
                <material name="gray"/>
            </visual>
            <collision>
                <geometry>
                    <cylinder radius="${(wheel_radius)}" length="${(wheel_length)}"/>
                </geometry>
                <origin xyz="0 0 0" rpy="0 0 0"/>
            </collision>
            <xacro:cylinder_inertia m="1.0" r="${wheel_radius}" h="${wheel_length}" 
                                    xyz="0 0 0" rpy="0 0 0" />
       </link> -->


</robot>